### PR TITLE
Issue #12557: Restore pitest.sh to ease operation on local

### PR DESCRIPTION
**Pull request by romani**
Issue #12557

### CODE REVIEW

File: .ci/pitest.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# Attention, there is no "-x" to avoid problems on CircleCI
+set -e
+
+case $1 in
+
+pitest-annotation \
+|pitest-ant \
+|pitest-api \
+|pitest-blocks \
```
Comment (Vyom-Yadav) [inline .ci/pitest.sh:10]: Can these not be hard coded and derived from `pom.xml`

File: .ci/pitest.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#   # post validation is skipped, we do not test gui throughly
+#   ;;
+
+--list)
+  echo "Supported profiles:"
+  pomXmlPath="$(dirname "${0}")/../pom.xml"
+  sed -n -e 's/^.*<id>\(pitest-[^<]\+\)<\/id>.*$/\1/p' < "${pomXmlPath}" | sort
+  ;;
+
+*)
```
Comment (Vyom-Yadav) [inline .ci/pitest.sh:51]: Why have separate logic for this? The groovy script has `--list` functionality.

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
The script has a `-g` flag too, please add that too.

File: .ci/pitest.sh (hunk 2 above)
Comment (romani) [reply .ci/pitest.sh:51]: it has bad output:
```
$ groovy ./.ci/pitest-survival-check-xml.groovy --list
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass$3$1 (file:/opt/groovy/groovy-2.4.7/lib/groovy-2.4.7.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass$3$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Supported profiles:
....
```

better to not bother user with all this noise.

File: .ci/pitest.sh (hunk 1 above)
Comment (romani) [reply .ci/pitest.sh]: I made bash more smart.

File: .ci/pitest.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+--list)
+  echo "Supported profiles:"
+  list_profiles "${0}"
+  ;;
+
+*)
+  PROFILES=$(list_profiles "${0}");
+
+  if [[ $(echo $PROFILES | grep -w -- ${1} | cat) != "" ]]; then
```
Comment (romani) [inline .ci/pitest.sh:20]: read 
 `--` - https://stackoverflow.com/a/2427987
`-w` - https://stackoverflow.com/a/8063284

Comment (Vyom-Yadav) [reply .ci/pitest.sh]: Isn't 
```bash
if [[ $(echo "$PROFILES" | grep -w "${1}") != "" ]]; then
```
sufficient?

`--` is not required as `$1` would be `pitest-XYZ` not something like `--pitest-XYZ`, also `cat` can also be omitted, it is already printed to output stream.

**Comment by Vyom-Yadav** [comment]
Please address https://github.com/checkstyle/checkstyle/pull/12561#pullrequestreview-1233328774 too

File: .ci/pitest.sh (hunk 2 above)
Comment (Vyom-Yadav) [reply .ci/pitest.sh]: Resolved

File: .ci/pitest.sh (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/pitest.sh]: Resolved

File: .ci/pitest.sh (hunk 3 above)
Comment (romani) [reply .ci/pitest.sh]: Without `--` , `--lis` is failing in weird way.

Cat is good to avoid failure is grep found nothing.

Comment (Vyom-Yadav) [reply .ci/pitest.sh]: Good points, resolved.

**Comment by romani** [comment]
I hope I fixed last violation in shell script, @Vyom-Yadav , please finish review.

**Comment by Vyom-Yadav** [comment]
Please add the `[-g | --generate-suppression]` flag, from the actual script:
```
Usage groovy ./.ci/pitest-survival-check-xml.groovy [profile] [-g | --generate-suppression]
To see the full list of supported profiles run
groovy ./.ci/ pitest-survival-check-xml.groovy --list
```

**Comment by romani** [comment]
I don't see it in GitHub action. Am I missing something?

**Comment by romani** [comment]
Script will be reused in both usages, did skip g on good reason or just missed?

**Comment by Vyom-Yadav** [comment]
> Script will be reused in both usages, did skip g on good reason or just missed?

Yes, it will be used for both, the `-g` flag is pretty helpful when you just want the suppressions.

**Comment by romani** [comment]
done

**Review by nrmancuso — APPROVED** [review]

**Comment by nrmancuso** [comment]
CI failure is in travis, it is well known.

**Pull request merged**